### PR TITLE
Refactoring and web3 verification

### LINKED ISSUE #89: Implement web3 identity verification
The verification logic itself should be a copy-paste of IMP mock, but we need to consider the code structure: some refactoring maybe needed.

### LINKED ISSUE #104: Define the `challenge_code` type in primitives
We should import the `challenge_code` type from primitives.

### LINKED ISSUE #105: Simplify verification logic
In #96  it was mentioned that we need to simplify the verification logic.

Currently logic:
https://github.com/litentry/tee-worker/blob/6cbfff69b3/litentry/core/identity-verify-handler/src/web2_identity/mod.rs#L182-L211

It should just compare the expected payload and the actual payload

### PR DESCRIPTION
**Pull request by Kailai-Wang**
resolves #89 
resolves #104 
resolves #105 

This is a big PR containing a lot of refactoring and cleanup:
- changes the ChallengeCode type to`[u8; 16]`
- adds web3ID verification logic
- adjusts web2ID verification logic
- refactoring, some ideas behind that:
   - have a separate `identity-verification` crate for verification stf tasks
   - have an empty `ruleset-build` crate for future ruleset building stf tasks
   - separate the TrustedCall submission (now called `StfTaskContext`) from the verification itself
   - HTTPClient usage should only be used in web2ID verification
   - create and only use the `StfTaskContext` from `stf_task_receiver`
   - eliminate common code and try to make it simple
   - cargo dep tidy-up
   - simplify trait bound in web2ID verification, only pass the required struct/ref

Hope that makes it a bit cleaner
 
Still many TODOs, mostly already written in the code:
- review/simplify the `Web2IdentityVerification` logic/struct definition
- review the mpsc channel creation
- error handling in many places
- consider if we should process all stf-tasks async
- fn name changes: #111 
- the verification is **not** yet integration-tested: #108

### CODE REVIEW

File: litentry/core/identity-verification/src/web2/mod.rs [hunk 1]
```diff
@@ -62,12 +62,45 @@ const TWITTER_BASE_URL: &str = "http://localhost";
...
 const DISCORD_BASE_URL: &str = "http://localhost";
 
+const TIMEOUT: Duration = Duration::from_secs(3u64);
+
+// TODO: maybe split this file into smaller mods
+
+#[derive(Debug, thiserror::Error, Clone)]
+pub enum Error {
+	#[error("Request error: {0}")]
+	RquestError(String),
```
Comment (ziming-zung) [inline litentry/core/identity-verification/src/web2/mod.rs]: `RquestError` typo error

File: app-libs/stf/src/stf_sgx_litentry.rs [hunk 2]
```diff
@@ -62,16 +56,26 @@ impl Stf {
...
+			identity: identity.clone(),
 			metadata,
 			linking_request_block: bn,
 		}
 		.dispatch_bypass_filter(ita_sgx_runtime::Origin::root())
 		.map_err(|e| StfError::Dispatch(format!("{:?}", e.error)))?;
-		// TODO: generate challenge code
+
+		// generate challenge code
+		let code = generate_challenge_code();
```
Comment (ziming-zung) [inline app-libs/stf/src/stf_sgx_litentry.rs:70]: I think we should return the `challenge_code` and let the parachain know that the `challenge_code` has been generated

**Review by ziming-zung — APPROVED** [review]
Looks good. Only two small things need to be modified.

File: app-libs/stf/src/stf_sgx_litentry.rs (hunk 2 above)
Comment (Kailai-Wang) [reply app-libs/stf/src/stf_sgx_litentry.rs:70]: Indeed, thanks!

**Pull request merged**